onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -5907,19 +5907,137 @@ def aten_std_mean(self: TensorType, unbiased: bool = True) -> tuple[TensorType,
+        right_win = op.Expand(op.Constant(value_ints=[0]), right)
+        right_win = op.CastLike(right_win, window)
+        left_win = op.CastLike(left_win, window)
+        window = op.Concat(left_win, window, right_win, axis=0)
+    return window
+
+
+@torch_op("aten::stft", private=True)
+def _create_window_from_win_length(win_length: int, n_fft: int) -> TFloatOrBFloat16:
+    left = (n_fft - win_length) / 2
+
+    right = n_fft - left - win_length
+    left = op.Reshape(left, op.Constant(value_ints=[1]))
+    right = op.Reshape(right, op.Constant(value_ints=[1]))
+    win_length = op.Reshape(win_length, op.Constant(value_ints=[1]))
+
+    left_win = op.Expand(op.Constant(value_ints=[0]), left)
+    right_win = op.Expand(op.Constant(value_ints=[0]), right)
+    window_list = op.Expand(op.Constant(value_ints=[1]), win_length)
+    return op.Concat(left_win, window_list, right_win, axis=0)
+
+
+@torch_op("aten::stft", private=True)
+def _create_window_from_n_fft(n_fft: int) -> TFloatOrBFloat16:
+    n_fft_tensor = op.Reshape(n_fft, op.Constant(value_ints=[1]))
+    window = op.Expand(op.Constant(value_ints=[1]), n_fft_tensor)
+    return window
+
+
+@torch_op("aten::stft", private=True)
+def _normalization(
+    signal: TFloatOrBFloat16, result: TFloatOrBFloat16, n_fft: int
+) -> TFloatOrBFloat16:
+    n_fft_tensor = op.Reshape(n_fft, op.Constant(value_ints=[1]))
+    sqrt_nfft = op.Sqrt(op.CastLike(n_fft_tensor, signal))
+    result = result / sqrt_nfft
+    return result
+
+
+@torch_op("aten::stft", private=True)
+def _aten_stft_onnx(
+    signal: TFloatOrBFloat16,
+    frame_step_const: INT64,
+    window: Union[TFloatOrBFloat16, INT64],
+    frame_length_const: INT64,
+    signal_rank: INT64,
+    onesided: int,
+) -> TFloatOrBFloat16:
+    window = op.CastLike(window, signal)
+    result = op.STFT(signal, frame_step_const, window, frame_length_const, onesided=onesided)
+    result = op.Transpose(result, perm=[0, 2, 1, 3])
+    # Remove batch dimension, if needed
+    if signal_rank == 1:
+        result = op.Squeeze(result, op.Constant(value_ints=[0]))
+    return result
+
+
+@torch_op("aten::stft", trace_only=True)
 def aten_stft(
-    self: TensorType,
+    self: TFloatOrBFloat16,
     n_fft: int,
     hop_length: Optional[int] = None,
     win_length: Optional[int] = None,
-    window: Optional[TensorType] = None,
+    window: Optional[TFloatOrBFloat16] = None,
     normalized: bool = False,
     onesided: Optional[bool] = None,
-    return_complex: Optional[bool] = None,
-) -> TensorType:
+    return_complex: Optional[bool] = None,  # pylint: disable=unused-argument
+) -> TFloatOrBFloat16:
     """stft(Tensor self, int n_fft, int? hop_length=None, int? win_length=None, Tensor? window=None, bool normalized=False, bool? onesided=None, bool? return_complex=None) -> Tensor"""
 
-    raise NotImplementedError()
+    # NOTE: regarless of the value of return_complex, we always return a real representation.
+
```

Comment:
We can do 

```python
del return_complex  # Unused
```

in trace only mode to avoid the pylint ignore.

Reply:
Thanks!